db/migrate/20190921232640_add_missing_run_fields_safety.rb

```diff
@@ -0,0 +1,7 @@
+class AddMissingRunFieldsSafety < ActiveRecord::Migration[6.0]
+  def change
+    change_column_null :run_histories, :pause_duration_ms, false
```

Comment:
I think we might need this to allow nulls here for timers / formats that don't report pause durations on run histories (such as the exchange format!).

Reply:
The parser defaults this to 0, which I think is better than accepting null because then you can perform operations on pause time without needing null checks.  What do you think about that?

Reply:
If I had the choice I'd choose for it to report null when pause time is not recorded (and 0 when it is but there were no pauses).

Similar situation to how we used to use 0 for absent run durations, segment durations, etc. -- it definitely makes the code easier for the reason you said, but that is a double-edged sword that also causes a lot of bugs that we've experienced over the years. e.g. summing a set containing 0 is fine, but averaging a set containing 0 throws the average way off. Charting 0 makes the chart wrong. And if we straight-up say "No pauses" on the run page, we'd be maybe lying.

But if the parser reports 0 for null then it sounds like we have no choice, so it's fine.

Reply:
No no no, I'm defaulting it to be 0, I think LSC reports it as null if it doesn't exist.  Your argument makes a lot of sense, so I will stop defaulting to 0 and allow nulls in the db.

Reply:
Ah ok gotcha, I thought you meant livesplit-core.